### PR TITLE
feat(cni): k8s make namespace configurable

### LINKED ISSUE #6660: CNI namespace should be configurable
### Description

Some cloud providers / platforms restrict certain namespaces. We should provide a way of configuring the namespace for CNI installation.

https://github.com/kumahq/kuma/blob/2ec76f34d58f88fd1d5326f5c89897e7dc7fb27e/pkg/plugins/runtime/k8s/controllers/cni_taint_controller.go#LL26C5-L26C5

**Note** that this warning from Istio applies to our CNI as well:

https://artifacthub.io/packages/helm/istio-official/cni/1.17.2#installing-the-chart

### PR DESCRIPTION
**Pull request by mmorel-35**
### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues -- closes #6660 
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by michaelbeaumont — DISMISSED** [review]

File: pkg/config/plugins/runtime/k8s/config.go [hunk 1]
```diff
@@ -298,6 +299,8 @@ type NodeTaintController struct {
 	Enabled bool `json:"enabled" envconfig:"kuma_runtime_kubernetes_node_taint_controller_enabled"`
 	// Value of app label on CNI pod that indicates if node can be ready.
 	CniApp string `json:"cniApp" envconfig:"kuma_runtime_kubernetes_node_taint_controller_cni_app"`
+	// Value of CNI pod namespace.
+	CniPodNamespace string `json:"cniPodNamespace" envconfig:"kuma_runtime_kubernetes_node_taint_controller_cni_pod_namespace"`
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/config/plugins/runtime/k8s/config.go:303]: Ah the node taint controller is running in the CP of course

**Comment by michaelbeaumont** [comment]
I think this will need a change to the CNI manifests in the helm chart as well:

https://github.com/kumahq/kuma/blob/193d7d7a619fb7eff2fbd1caffd3f370098c3feb/deployments/charts/kuma/templates/cni-daemonset.yaml#L6

File: deployments/charts/kuma/values.yaml [hunk 2]
```diff
@@ -306,6 +306,8 @@ cni:
   tolerations: []
   # -- Additional pod annotations
   podAnnotations: { }
+  # -- Pod Namespace 
+  podNamespace: kube-system
```
Comment (michaelbeaumont) [inline deployments/charts/kuma/values.yaml:310]:
```suggestion
  namespace: kube-system
```

File: pkg/config/plugins/runtime/k8s/config.go (hunk 1 above)
Comment (michaelbeaumont) [inline pkg/config/plugins/runtime/k8s/config.go:303]:
```suggestion
	CniNamespace string `json:"cniNamespace" envconfig:"kuma_runtime_kubernetes_node_taint_controller_cni_namespace"`
```

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

**Review by michaelbeaumont — APPROVED** [review]
LGTM, @bartsmykla can you take a look?

**Review by bartsmykla — APPROVED** [review]

**Comment by bartsmykla** [comment]
Thank you @mmorel-35 for the contribution!

**Pull request merged**